### DISCUSSION TITLE
nucleo-f413zh REPL over uart

### DISCUSSION THREAD

**Discussion by crsrinidhi** [micropython/micropython#9510] [Hardware & Peripherals]
Hi,

I have a NUCLEO-F413ZH board from which i have cut off the stlink part of the board.

To program the board I connect the stlink and the main board with jumper wires and program. This works correctly. 

I want to use this board with micropython and have used the latest micropython for the board from the page in the link below flashed it using the stlink connection.

https://micropython.org/download/NUCLEO_F413ZH/

I have tried the latest stable release and the latest nightly builds.

The flash is successful without any errors. But I am unable to access the REPL terminal using any of the UART on the board.

From the mpconfiguration.h file the REPL uart is the PYB_UART_3, with the baud rate being 115200.

https://github.com/micropython/micropython/blob/master/ports/stm32/boards/NUCLEO_F413ZH/mpconfigboard.h

I am using a usb to serial adapter for connecting the board to the laptop. Even with the crosswiring of the RX and TX the REPL is not accessible.

Any pointers to access the REPL prompt will be helpful. Thanks in advance.

**Comment by dpgeorge** [comment]
The REPL is also on the USB on the main part of the board (not the stlink part).  Did you try that?

Failing that, you can freeze in a main.py which blinks the LED to test that it is actually running MicroPython.

**Reply by crsrinidhi** [reply]
Before seperating the stlink form the main board I used to connect the power to the stlink and use the USB for accessing the REPL.
After seperating the stlink i am not able to access the REPL on the USB, even on the pins PA11 and PA12 pins the REPL is not accessible. So i was looking at alternative methods for REPL access.

**Reply by dpgeorge** [reply]
> Before seperating the stlink form the main board I used to connect the power to the stlink and use the USB for accessing the REPL.

Did you use the stlink USB port for the REPL, or the other USB port at the other end of the Nucleo board (connected to PA11/PA12)?

It may be you damaged something during separation.  Or maybe the stlink is providing a clock source for the main MCU, which is now missing.

**Reply by crsrinidhi** [reply]
I used the stlink USB port to power the board and the other USB to access the REPL. It was working fine. Then I separated the stlink and may be it is missing a signal for the USB. If I program the board like an arduino using the stlink with jumper wires it works correctly. I will have to go through the manual again carefully.

**Reply by crsrinidhi** [reply]
> > Before seperating the stlink form the main board I used to connect the power to the stlink and use the USB for accessing the REPL.
> 
> Did you use the stlink USB port for the REPL, or the other USB port at the other end of the Nucleo board (connected to PA11/PA12)?
> 
> It may be you damaged something during separation. Or maybe the stlink is providing a clock source for the main MCU, which is now missing.

Will try to solder the stlink back to the main board and try. Thank you for your time.
